site/src/nav/DownloadButton.jsx

```diff
@@ -42,49 +42,62 @@ function DownloadButton({ mode, zoom, setZoom }) {
     console.log(bounds);
 
     //Send those to the download engine
-    const minxPath = ["bbox", "minx"];
-    const minyPath = ["bbox", "miny"];
-    const maxxPath = ["bbox", "maxx"];
-    const maxyPath = ["bbox", "maxy"];
+    const xmin = ["bbox", "xmin"];
+    const ymin = ["bbox", "ymin"];
+    const xmax = ["bbox", "xmax"];
+    const ymax = ["bbox", "ymax"];
 
     const readOptions = {
       bbox: bbox,
       bboxPaths: {
-        minxPath,
-        minyPath,
-        maxxPath,
-        maxyPath,
+        xmin,
+        ymin,
+        xmax,
+        ymax,
       },
     };
-
-    let parquetDataset = await new ParquetDataset(DownloadCatalog);
-    const wasmTable = await parquetDataset.read(readOptions);
+    let downloadCatalog = getDownloadCatalog(bbox, visibleTypes);
 
     set_panic_hook();
-    //Create a blob
-    // const binaryDataForDownload = writeGeoParquet(wasmTable);
-    const binaryDataForDownload = writeGeoJSON(wasmTable);
 
-    let blerb = new Blob([binaryDataForDownload], {
-      type: "application/octet-stream",
+    // The catalog contains a base path and then a list of types with filenames. 
+    //First, assemble the parquet datasets in parallel.
+   let datasets = downloadCatalog.types.map(type => {
+      return new ParquetDataset(downloadCatalog.basePath, type.files).then((dataset) => {return {type: type.name, parquet: dataset} });
     });
+      
+    Promise.all(datasets)
+    .then((datasets) => {
+      return datasets.map(dataset => dataset.parquet.read(readOptions).then(reader => {return {type: dataset.type, reader:reader}}));
+    }).then((tableReads) => Promise.all(tableReads)
+    .then((wasmTables) => {
```

Comment:
Should we filter out the returned features which don't intersect with the viewport bbox? I believe we currently are downloading the entire parquet file as geojson if there's any overlap.

Reply:
Ah! Yes, that is part of the 'readOptions' that get fed into the geoarrow code- all we've done till now is broadly select which geoparquet files to pick from, but the readoptions further pare down the data that we send as part of the download.